custom_components/plugwise/config_flow.py

```diff
@@ -334,27 +335,27 @@ async def async_step_init(
             return self.async_create_entry(title="", data=user_input)
 
         coordinator = self.hass.data[DOMAIN][self.config_entry.entry_id][COORDINATOR]
-        interval = DEFAULT_SCAN_INTERVAL[coordinator.api.smile_type]
+        interval = DEFAULT_SCAN_INTERVAL[coordinator.api.smile_type]  # pw-beta
 
         data = {
             vol.Optional(
-                CONF_SCAN_INTERVAL,
+                timedelta(seconds=CONF_SCAN_INTERVAL),
                 default=self.config_entry.options.get(CONF_SCAN_INTERVAL, interval),
```

Comment:
Shouldn't this be `default = timedelta(self.config_entry.options.get(CONF_SCAN_INTERVAL, interval)),`

Reply:
good catch, apparently not caught by flake/mypy

Reply:
changed accordingly (and simplified)